Source/WebKit/UIProcess/_WKWebViewPrintFormatter.mm

```diff
@@ -66,6 +71,33 @@ - (WKWebView *)_webView
     return static_cast<WKWebView *>(view);
 }
 
+- (CGPDFDocumentRef)_printedDocument
+{
+    if (self.requiresMainThread)
+        return _printedDocument.get();
+
+    Locker locker { _printLock };
+    return _printedDocument.get();
+}
+
+- (void)_setPrintedDocument:(CGPDFDocumentRef)printedDocument
+{
+    if (self.requiresMainThread) {
+        _printedDocument = printedDocument;
+        return;
+    }
+
+    Locker locker { _printLock };
+    _printedDocument = printedDocument;
+    _printCompletionCondition.notifyOne();
```

Comment:
do we also need to do this in a `dealloc` or something in case `-[WKContentView _resetPrintingState]` is called?

Reply:
UIKit will keep the print formatter alive until printing is completed, so it cannot be deallocated when waiting.

`-[WKContentView _resetPrintingState]` will unblock the thread in the event of web process termination.